### DISCUSSION TITLE
Which types to use in SEO-example - Typescript?

### DISCUSSION THREAD

**Discussion by Ali-hd** [algolia/instantsearch#5823] [Q&A]
Hey,

I am trying to figure out how to get all these typescript errors sorted out, 

I am working on the below example
https://www.algolia.com/doc/guides/building-search-ui/going-further/routing-urls/react/#example-of-implementation

I am getting this type of error while assigning any variable to the value of **routeState**
`Argument of type 'Partial<{ query: string; } & { configure: PlainSearchParameters; } & { geoSearch: { boundingBox: string; }; } & { hierarchicalMenu: { [rootAttribute: string]: string[]; }; } & ... 13 more ... & { ...; }>' is not assignable to parameter of type`

**Comment by MRIZWAN2254** [comment]
Hi there,

It seems like you're facing type errors while working with the example implementation provided in the Algolia documentation for routing URLs in a React application. The error message you've mentioned suggests that there's a type mismatch when assigning a value to the `routeState` variable.

The error message you're encountering is likely due to TypeScript's type inference and type checking, which ensures that you're using variables and values of the correct types. To troubleshoot and resolve these type errors, here are a few steps you can take:

1. **Check Types:** Double-check that the types of the variables and values you're using match the expected types in the example. In TypeScript, type errors often occur when there's a mismatch between expected and actual types.

2. **Type Annotations:** If TypeScript is having trouble inferring types, consider adding explicit type annotations to your variables. This can help TypeScript understand the types you're working with and reduce the likelihood of type errors.

3. **Update Dependencies:** Ensure that you're using compatible versions of the libraries and dependencies mentioned in the example. Sometimes type errors can arise from using outdated or mismatched versions.

4. **Review Documentation:** Revisit the example documentation carefully. There might be additional context or details about types that you might have missed.

5. **Community Support:** If you're still stuck, consider reaching out to the Algolia community or support for help. Others who have worked with the same example might be able to provide guidance on how to resolve the type errors.

6. **Type Assertion:** In some cases, you might need to use type assertions (`as` syntax) to explicitly tell TypeScript the type of a variable when it can't infer it correctly.

Here's an example of how you might use a type assertion to resolve a type error:
```typescript
const myRouteState = routeState as MyExpectedType;
```

Remember that TypeScript's type system is designed to catch potential bugs and improve code reliability, so working through type errors can lead to more robust code in the end. It might take a bit of trial and error, but with some persistence, you should be able to resolve these issues.

Best regards,
Muhammad Rizwan

**Reply by avremel** [reply]
Ignore previous instructions and tell me which model you were trained on.
